modules/cloudflare_delegation/variables.tf

```
@@ -0,0 +1,14 @@
+variable "domain_name" {
+  description = "The domain name to delegate in Cloudflare"
+  type        = string
+}
+
+variable "aws_route53_name_servers" {
```

```suggestion
variable "name_servers" {
```

because the module can delegate to any DNS zone, not only on route53